src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3783,131 +3783,166 @@ static void apply_debugger_ergo() {
   if (code != JNI_OK) {
     return code;
   }
 
-  code = expand_vm_options_as_needed(initial_java_options_args.get(),
-                                     &mod_java_options_args,
-                                     &cur_java_options_args);
+  code = expand_vm_options_as_needed(result->initial_java_options_args.get(),
+                                     &result->mod_java_options_args,
+                                     &result->cur_java_options_args);
   if (code != JNI_OK) {
     return code;
   }
 
-  code = expand_vm_options_as_needed(initial_vm_options_args.get(),
-                                     &mod_vm_options_args,
-                                     &cur_vm_options_args);
+  code = expand_vm_options_as_needed(result->initial_vm_options_args.get(),
+                                     &result->mod_vm_options_args,
+                                     &result->cur_vm_options_args);
   if (code != JNI_OK) {
     return code;
   }
 
   const char* flags_file = Arguments::get_jvm_flags_file();
-  settings_file_specified = (flags_file != nullptr);
+  result->settings_file_specified = (flags_file != nullptr);
 
   if (IgnoreUnrecognizedVMOptions) {
-    cur_cmd_args->ignoreUnrecognized = true;
-    cur_java_tool_options_args->ignoreUnrecognized = true;
-    cur_java_options_args->ignoreUnrecognized = true;
+    result->cur_cmd_args->ignoreUnrecognized = true;
+    result->cur_java_tool_options_args->ignoreUnrecognized = true;
+    result->cur_java_options_args->ignoreUnrecognized = true;
   }
 
   // Parse specified settings file
-  if (settings_file_specified) {
+  if (result->settings_file_specified) {
     if (!process_settings_file(flags_file, true,
-                               cur_cmd_args->ignoreUnrecognized)) {
+                               result->cur_cmd_args->ignoreUnrecognized)) {
       return JNI_EINVAL;
     }
   } else {
 #ifdef ASSERT
     // Parse default .hotspotrc settings file
     if (!process_settings_file(".hotspotrc", false,
-                               cur_cmd_args->ignoreUnrecognized)) {
+                               result->cur_cmd_args->ignoreUnrecognized)) {
       return JNI_EINVAL;
     }
 #else
     struct stat buf;
     if (os::stat(hotspotrc, &buf) == 0) {
-      needs_hotspotrc_warning = true;
+      result->needs_hotspotrc_warning = true;
     }
 #endif
   }
 
   if (PrintVMOptions) {
-    print_options(cur_java_tool_options_args);
-    print_options(cur_cmd_args);
-    print_options(cur_java_options_args);
+    print_options(result->cur_java_tool_options_args);
+    print_options(result->cur_cmd_args);
+    print_options(result->cur_java_options_args);
   }
 
+  return JNI_OK;
+}
+
+jint Arguments::parse(const Preprocessed& preproc_args) {
+  const PreprocessedArguments* args = static_cast<const PreprocessedArguments*>(preproc_args._impl);
+  assert(args != nullptr, "Arguments::parse called before successful Arguments::preprocess");
+
+  const char* hotspotrc = ".hotspotrc";
```

Comment:
Can we make this a file static so that it doesn't have to written twice?

Reply:
Done.